Model/ModelManager.php

```diff
@@ -92,13 +94,26 @@ public function getParentMetadataForProperty($baseClass, $propertyFullName)
                 $parentAssociationMappings[] = $metadata->associationMappings[$nameElement];
                 $class = $metadata->getAssociationTargetClass($nameElement);
             } elseif (isset($metadata->embeddedClasses[$nameElement])) {
-                $parentAssociationMappings = array();
-                $lastPropertyName = $propertyFullName;
-                $class = $baseClass;
+                $parentEmbeddedMappings[] = array(
+                    'class' => $class,
+                    'field' => $nameElement,
+                );
+                $class = $metadata->embeddedClasses[$nameElement]['class'];
             }
         }
 
-        return array($this->getMetadata($class), $lastPropertyName, $parentAssociationMappings);
+        // Support for multi-level embedded properties
+        if (!empty($parentEmbeddedMappings)) {
+            $class = $parentEmbeddedMappings[0]['class'];
```

Comment:
Please use `current` over `[0]`, it is more robust.

Reply:
Done